src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs

```diff
@@ -14,6 +18,73 @@ internal static void VerifyValidPath(string path, string argName)
             }
         }
 
+        internal static bool Copy(string sourcePath, string destinationPath, bool recursive,
+            bool skipExistingFiles, CancellationToken cancellationToken)
+        {
+            SearchOption searchOption = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            IEnumerable<string> directoryEnumeration;
+            bool success = true;
+
+            try
+            {
+                EnumerationOptions enumerationOptions = EnumerationOptions.Compatible;
+
+                // Compatible uses `IgnoreInaccessible = false`
+                enumerationOptions.IgnoreInaccessible = true;
+
+                if (recursive)
+                {
+                    enumerationOptions.RecurseSubdirectories = true;
```

Comment:
Now it won't affect `EnumerationOptions.Compatible` and the same tests are failing.

Reply:
It is still affecting since you are holding a reference to it in line 30.
https://github.com/dotnet/runtime/blob/ce89d2de42483c5d1363d35e141289b9d42f81ff/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs#L30

Reply:
@Jozkee Oh you're right. I see `RecurseSubdirectories`...

Reply:
All right. Thanks for the clarification.

I have the native approach on the screen. I will investigate the benchmarking topic in more detail tomorrow.

Reply:
Oh yes, you can re-use those. Good point :)

Reply:
CI is passing now.